documentation/adr/0018-algorithm-migration-to-paged-files.md

```diff
@@ -0,0 +1,115 @@
+[<-previous](0015-store-pixel-data-in-single-precision.md) | next->
+
+# 18 - Algorithm migration to paged implementations
+
+Date: 2020-Oct-15
+
+## Status
+
+Accepted
+
+## Context
+
+The codebase includes two different `cut()` methods:
+ - one in the SQW object API which operates on the in-memory SQW data
+ - one is a function that takes a `.sqw` filename and operates directly on that (on-disk)
+
+The on-disk implementation is significantly slower and users are aware of this.
+If a 'too large' file is operated on in-memory an OutOfMemory Error will be thrown by the MATLAB UI or backend and the process fail. 
+The typical use-case for cut is that a initial cut is made from the complete dataset in a file into an object which is small and fits in memory.
+Subsequent cuts are in-memory object to in-memory object.
+
+As a consequence:
+ - users are required to develop (and test) separate scripts for each API
+ - users make a conscious choice between “in memory (fast)” and “on disk (slow)” operation.
+
+Mantid have (multiple times) attempted to present seamless in-memory/on-disk operation to their users (for a slightly different use case).
+Each time this either failed for technical reasons or because users did not embrace it,
+and the attempts were abandoned.
+
+One of driving forces for the current PACE project is the support of large datasets,
+which are expected to be increasingly common and of increasing size.
+The new SQW object hides all choice between in-memory and on-disk operation through
```

Comment:
It doesn't yet. At the moment the pixel page size is large enough such that everything is always in memory. Is this an aspiration?